packages/ui/src/dictionary-controller/DictionaryDataContext.tsx

```diff
@@ -0,0 +1,165 @@
+	setFilters: (filters: FilterOptions[]) => void;
+};
+
+export type StaticDictionaryProviderProps = {
+	children: ReactNode;
+	staticDictionaries: DictionaryServerUnion[];
+};
+
+export type RemoteDictionaryProviderProps = {
+	children: ReactNode;
+	lecternUrl: string;
+	dictionaryName: string;
+};
+
+export type HostedDictionaryProviderProps = {
+	children: ReactNode;
+	hostedUrl: string;
+};
+
+export type DictionaryProviderProps =
+	| StaticDictionaryProviderProps
+	| RemoteDictionaryProviderProps
+	| HostedDictionaryProviderProps;
+
+const DictionaryDataContext = createContext<DictionaryContextType | undefined>(undefined);
+
+export function useDictionaryDataContext(): DictionaryContextType {
+	const context = useContext(DictionaryDataContext);
+	if (context === undefined) {
+		console.error('useDictionaryDataContext must be used within a DictionaryDataProvider');
+		throw new Error('useDictionaryDataContext must be used within a DictionaryDataProvider');
+	}
+	return context;
+}
+
+export function DictionaryDataProvider(props: DictionaryProviderProps) {
+	const [dictionaries, setDictionaries] = useState<DictionaryServerUnion[]>([]);
+	const [loading, setLoading] = useState(true);
+	const [errors, setErrors] = useState<string[]>([]);
+
+	const [currentDictionaryIndex, setCurrentDictionaryIndex] = useState(0);
+	const [filters, setFilters] = useState<FilterOptions[]>([]);
+
+	useEffect(() => {
+		if ('staticDictionaries' in props) {
+			setDictionaries(props.staticDictionaries);
+			setLoading(false);
+			setErrors([]);
+			return;
+		}
+
+		if ('hostedUrl' in props) {
+			const fetchHostedDictionaries = async () => {
+				try {
+					const dictionariesData = await fetchAndValidateHostedDictionaries(props.hostedUrl);
+					setDictionaries([dictionariesData]);
+					setErrors([]);
+				} catch (err) {
+					console.error('Error loading hosted dictionary data:', err);
+					let message: string;
+					switch (true) {
+						case err instanceof Error:
+							message = err.message;
+							break;
+						case typeof err === 'string':
+							message = err;
+							break;
+						default:
+							message = 'Something went wrong';
+					}
+					setErrors([message]);
+				} finally {
+					setLoading(false);
+				}
+			};
+			fetchHostedDictionaries();
+			return;
+		}
+
+		if ('lecternUrl' in props) {
+			const fetchRemoteDictionaries = async () => {
+				try {
+					const { dictionaries: fetchedDictionaries } = await fetchRemoteDictionary(
+						props.lecternUrl,
+						props.dictionaryName,
+					);
+					setDictionaries(fetchedDictionaries);
+					setErrors([]);
+				} catch (err) {
+					console.error('Error loading hosted dictionary data:', err);
+					let message: string;
+					switch (true) {
+						case err instanceof Error:
+							message = err.message;
+							break;
+						case typeof err === 'string':
+							message = err;
+							break;
+						default:
+							message = 'Something went wrong';
+					}
```

Comment:
`switch(true)` is an antipattern
the `case` is meant to be the conditional you're checking in the switch(..) call
using it this way is a more obtuse if/else... so you can just use an if / else ;)